### DISCUSSION TITLE
Error on browser.url trying to open app in localhost:3000

### DISCUSSION THREAD

**Discussion by mpineyrozaelot** [webdriverio/webdriverio#10485] [Q&A] [answered]
Good morning community! How are you?
I started to use WebDriverIO 5 years ago, and I never had issues actually, but today I have one that I don't know how to solve, maybe you can help me.
I'm deploying an application locally which is hosted in localhost:3000, and I'm trying to point my tests to this environment.
For some reason the test is failing after : 

`await browser.url("");`

And the error message is : 

**ERROR webdriver: Request failed with status undefined due to TimeoutError: Timeout awaiting 'response' for 30000ms when running "url" with method "POST" and args "{"url":"http://localhost:3000/"}"**

The weirdest part is that the page is being opened correctly, so I don't know why it fails. This is not happening in other environments, just in localhost.
Yesterday I updated all the packages, so I have the latest versions, these are my dependencies : 

"@faker-js/faker": "7.6.0",
    "@types/jasmine": "4.3.1",
    "@types/node": "18.11.18",
    "@typescript-eslint/eslint-plugin": "5.49.0",
    "@wdio/cli": "8.10.5",
    "@wdio/dot-reporter": "8.1.2",
    "@wdio/jasmine-framework": "8.10.5",
    "@wdio/junit-reporter": "8.2.0",
    "@wdio/local-runner": "8.10.5",
    "@wdio/runner": "8.10.6",
    "@wdio/selenium-standalone-service": "8.10.4",
    "@wdio/spec-reporter": "8.1.2",
    "@wdio/types": "8.3.0",
    "eslint": "8.32.0",
    "eslint-config-prettier": "8.6.0",
    "eslint-plugin-prettier": "4.2.1",
    "eslint-plugin-wdio": "8.0.14",
    "log4js": "6.7.1",
    "prettier": "2.8.3",
    "supertest": "6.3.3",
    "test-data-provider": "1.0.3",
    "ts-node": "10.9.1",
    "typescript": "5.0.4",
    "wdio-docker-service": "3.2.1",
    "wdio-eslinter-service": "0.0.4",
    "wdio-html-nice-reporter": "8.1.0",
    "wdio-video-reporter": "3.5.0",
    "wdio-wait-for": "3.0.1",
    "webdriverio": "8.10.5",
    "moment": "2.29.4",
    "wdio-intercept-service": "4.4.0",
    "aws-sdk": "2.1362.0",
    "@aws-sdk/client-ec2-instance-connect": "3.316.0"

I'm using TypeScript and Node 19.4.0.
The local application uses Node 14.21.2.

Any idea? Thanks for your help!

**Comment by christian-bromann** [comment]
Hard to say what is going on, can you provide a reproducible example?

**Reply by mpineyrozaelot** [reply]
I know, but no, I can't provide any reproducible example as it's just happening in localhost and I can't share the application.

**Comment by mpineyrozaelot** [comment, marked as answer]
Closing this ticket. I finally solved it with a workaround : 

- I enter to a wrong URL in the same domain (ex: /fakepage)
- I click on a button that redirects me to the page that I want to access

From that point I can continue the test execution.
Thanks!
